File: app/src/main/java/com/example/template/adapter/MenuAdapter.java
package com.example.template.adapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.example.template.R;
import com.example.template.bean.HomeMenuBean;

import java.util.List;

/**
 * 功能描述：主页RecycleView adapter
 * Created by gfq on 2020/3/26.
 */
public class MenuAdapter extends BaseQuickAdapter<HomeMenuBean, BaseViewHolder> {

    public MenuAdapter(@Nullable List<HomeMenuBean> data) {
        super(R.layout.item_menu, data);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, HomeMenuBean item) {
        helper.setText(R.id.tv_menu_name, item.getName());
    }
}

File: appkotlin/src/main/java/com/example/appkotlin/adapter/MenuAdapter.kt
package com.example.appkotlin.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.example.appkotlin.R
import com.example.appkotlin.bean.HomeMenuBean

/**
 *
 * 功能描述：
 * Created by gfq on 2020/12/1
 *
 **/

class MenuAdapter(data: List<HomeMenuBean>?) : BaseQuickAdapter<HomeMenuBean, BaseViewHolder>(R.layout.item_menu, data) {

    override fun convert(helper: BaseViewHolder, item: HomeMenuBean) {
        helper.setText(R.id.tv_menu_name, item.name)
    }
}